portfolio/src/main/webapp/script.js

```diff
@@ -25,3 +25,25 @@ setInterval(() => {
   }
 }, speed);
 
+function getGreeting() {
+  console.log('Fetching greeting');
+
+  const greetingPromise = fetch('/data');
+
+  greetingPromise.then(responseHandler);
+}
+
+function responseHandler(response) {
+  console.log('Handling response');
+
+  const textPromise = response.text();
+
+  textPromise.then(addGreetingToDom);
+}
```

Comment:
Consider to use async function to get rid of nested promise-then.

Reply:
What is the issue with a nested promise-then? I'm still learning how promises work more in-depth while I go through the tutorial.

Reply:
I see. If you want to practice the promise, go for the existing one.

One downside of the nested promise is that it makes the code difficult to understand. You will have to go into `responseHandler` then `addGreetingToDom` to understand how exactly the code handles the response.
If you use async function, you can shorten them into 3 lines and understand them in seconds.

The good part of promise is that you can define your error handling in each function.

Reply:
Ah I see, there's extra control for error handling at each step but to understand the process you have to consult three separate functions. The tutorial also suggested using arrow functions to shorten the promise chain, what is your take on this?

Thank you for the tips!

Reply:
Yes, arrow functions in js or lambda expression in java are widely used and is a good style to make the code more compact and readable. 
```
doA().then((resultA)=>{
  doB(resultA).then((resultB)=>{
    doC(resultB).then((resultC)=>{...})
  })
})
```

I personally still prefer async function when possible.
```
const resultA = async doA();
const resultB = async doB(resultA);
const resultC = async doC(resultB);
...
```

Reply:
Sounds good, thank you again Charlie!